api/osrd_infra/schemas/rolling_stock.py

```diff
@@ -29,9 +28,47 @@ class EffortCurve(BaseModel, extra=Extra.forbid):
     speeds: conlist(confloat(ge=0), min_items=2)
     max_efforts: conlist(confloat(ge=0), min_items=2)
 
-    @root_validator
+    @root_validator(skip_on_failure=True)
     def check_size(cls, v):
-        assert len(v.get("speeds")) == len(v.get("max_efforts")), "speeds and max_efforts must be the same length"
+        assert len(v["speeds"]) == len(v["max_efforts"]), "speeds and max_efforts must be the same length"
+        return v
+
+
+class EffortCurveConditions(BaseModel, extra=Extra.forbid):
+    comfort: Optional[Union[Literal["ac"], Literal["heating"]]]
+
+
+class ConditionalEffortCurve(BaseModel, extra=Extra.forbid):
+    """Effort curve subject to application conditions"""
+
+    cond: EffortCurveConditions = Field(default=EffortCurveConditions())
+    curve: EffortCurve = Field(description="Effort curve to apply if the conditions are met")
```

Comment:
In the future we will have multiple condition in `EffortCurveConditions` all of them being optional. However, having a `null` value for `cond` doesn't make sense since it's the same as using the `default_curve`.